service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/update/UpdateGcsBucketStep.java

```diff
@@ -0,0 +1,110 @@
+package bio.terra.workspace.service.resource.controlled.flight.update;
+
+import static bio.terra.workspace.service.resource.controlled.GcsApiConversions.toGcsApi;
+import static bio.terra.workspace.service.resource.controlled.GcsApiConversions.toGcsApiRulesList;
+import static bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys.PREVIOUS_UPDATE_PARAMETERS;
+import static bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys.UPDATE_PARAMETERS;
+
+import bio.terra.cloudres.google.storage.BucketCow;
+import bio.terra.cloudres.google.storage.StorageCow;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.generated.model.ApiGcpGcsBucketUpdateParameters;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.resource.controlled.ControlledGcsBucketResource;
+import bio.terra.workspace.service.workspace.WorkspaceService;
+import com.google.cloud.storage.BucketInfo.LifecycleRule;
+import com.google.cloud.storage.StorageClass;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UpdateGcsBucketStep implements Step {
+  private final Logger logger = LoggerFactory.getLogger(UpdateGcsBucketStep.class);
+  private final ControlledGcsBucketResource bucketResource;
+  private final CrlService crlService;
+  private final WorkspaceService workspaceService;
+
+  public UpdateGcsBucketStep(
+      ControlledGcsBucketResource bucketResource,
+      CrlService crlService,
+      WorkspaceService workspaceService) {
+    this.bucketResource = bucketResource;
+    this.crlService = crlService;
+    this.workspaceService = workspaceService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+    final FlightMap inputMap = flightContext.getInputParameters();
+    final ApiGcpGcsBucketUpdateParameters updateParameters =
+        inputMap.get(UPDATE_PARAMETERS, ApiGcpGcsBucketUpdateParameters.class);
+
+    return updateBucket(updateParameters);
+  }
+
+  // Restore the previous values of the update parameters
+  @Override
+  public StepResult undoStep(FlightContext flightContext) throws InterruptedException {
+    final FlightMap workingMap = flightContext.getWorkingMap();
+    final ApiGcpGcsBucketUpdateParameters previousUpdateParameters =
+        workingMap.get(PREVIOUS_UPDATE_PARAMETERS, ApiGcpGcsBucketUpdateParameters.class);
+
+    return updateBucket(previousUpdateParameters);
+  }
+
+  private StepResult updateBucket(ApiGcpGcsBucketUpdateParameters updateParameters) {
```

Comment:
I think the `updateParameters` can be null. Have you tested the case where the update request does not contain any update parameters at all?